File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2021s/IZ08IO1/Builder/src/IdBuilder.java

import java.util.HashSet;
import java.util.stream.Collectors;

public class IdBuilder
{
   //... where id in (1, 13, 666)
   //... where id in (null)
   //IdBuilder ib=new IdBuilder(); ib.setIfEmpty("").append(1).append(13).toString();
   private String Delimeter=", ";
   private String IfEmpty="null";
   private final HashSet<Long> list=new HashSet<>();

   public IdBuilder append(long id)
   {
      list.add(id);
      return this;
   }

   public IdBuilder clear()
   {
      list.clear();
      return this;
   }

   public IdBuilder setDelimeter(String Delimeter)
   {
      this.Delimeter=Delimeter;
      return this;
   }

   public IdBuilder setIfEmpty(String IfEmpty)
   {
      this.IfEmpty=IfEmpty;
      return this;
   }

   @Override public String toString()
   {
      if(list.isEmpty()) return IfEmpty;
      StringBuilder sb=new StringBuilder();
      boolean first=true;
      for(long id:list.stream().sorted().collect(Collectors.toList()))
      {
         if(first) first=false; else sb.append(Delimeter);
         sb.append(id);
      }
      return sb.toString();
   }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2021s/Id06IO2/Builder/Builder/IdBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Builder
{
    // ... where id in (5, 13, 99);
    // ... where id in (null);
    // idb;
    // idb.Append(5).Append(13);
    class IdBuilder
    {
        private List<long> array=new List<long>();
        private string IfEmpty="null";
        private string Delimeter=", ";
        public IdBuilder SetIfEmpty(string IfEmpty)
        {
            this.IfEmpty=IfEmpty;
            return this;
        }
        public IdBuilder SetDelimeter(string Delimeter)
        {
            this.Delimeter=Delimeter;
            return this;
        }
        public IdBuilder Append(long id)
        {
            array.Add(id);
            return this;
        }
        public override string ToString()
        {
            if (array.Count <= 0) return IfEmpty;
            StringBuilder sb=new StringBuilder();
            bool first=true;
            foreach(long id in array)
            {
                if(first) first=false; else sb.Append(Delimeter);
                sb.Append(id);
            }
            return sb.ToString();
        }
    }
}
